tools/cldr-apps/js/src/esm/cldrForum.js

```diff
@@ -373,68 +374,73 @@ function submitPost(event) {
-
-  const subj = document.getElementById("postSubject").innerHTML;
+function getFormValues(text) {
+  return {
+    text: text,
+    locale: $("#post-form input[name=_]").val(),
+    open: $("#post-form input[name=open]").val(),
+    postType: $("#post-form input[name=postType]").val(),
+    replyTo: $("#post-form input[name=replyTo]").val(),
+    root: $("#post-form input[name=root]").val(),
+    value: $("#post-form input[name=value]").val(),
+    xpath: $("#post-form input[name=xpath]").val(),
+  };
+}
 
+function sendPostRequest(form) {
   const url = cldrStatus.getContextPath() + "/SurveyAjax";
-
-  const errorHandler = function (err) {
-    const post = $(".post").first();
-    post.before("<p class='warn'>error! " + err + "</p>");
-  };
-  const loadHandler = function (data) {
-    if (data.err) {
-      const post = $(".post").first();
-      post.before("<p class='warn'>error: " + data.err + "</p>");
-    } else if (data.ret && data.ret.length > 0) {
-      const postModal = $("#post-modal");
-      postModal.modal("hide");
-      const curSpecial = cldrStatus.getCurrentSpecial();
-      if (curSpecial && curSpecial === "forum") {
-        cldrLoad.reloadV();
-      } else {
-        cldrForumPanel.updatePosts(null);
-      }
-    } else {
-      const post = $(".post").first();
-      post.before(
-        "<i>Your post was added, #" +
-          data.postId +
-          " but could not be shown.</i>"
-      );
-    }
-  };
   const postData = {
-    s: cldrStatus.getSessionId(),
-    _: locale,
-    replyTo: replyTo,
-    xpath: xpath,
-    text: text,
-    subj: subj,
-    postType: postType,
-    value: value,
-    open: open,
-    root: root,
     what: "forum_post",
+    s: cldrStatus.getSessionId(),
+    subj: document.getElementById("postSubject").innerHTML,
+    _: form.locale,
+    open: form.open,
+    postType: form.postType,
+    replyTo: form.replyTo,
+    root: form.root,
+    text: form.text,
+    value: form.value,
+    xpath: form.xpath,
   };
   const xhrArgs = {
     url: url,
     handleAs: "json",
-    load: loadHandler,
-    error: errorHandler,
+    load: loadHandlerForSubmit,
+    error: errorHandlerForSubmit,
     postData: postData,
   };
   cldrAjax.sendXhr(xhrArgs);
 }
 
+function loadHandlerForSubmit(data) {
+  if (data.err) {
+    const post = $(".post").first();
+    post.before("<p class='warn'>error: " + data.err + "</p>");
+  } else if (data.ret && data.ret.length > 0) {
+    const postModal = $("#post-modal");
+    postModal.modal("hide");
+    if (cldrStatus.getCurrentSpecial() === "forum") {
+      cldrLoad.reloadV();
+    } else {
+      cldrForumPanel.updatePosts(null);
+    }
+  } else {
+    const post = $(".post").first();
+    post.before(
+      "<i>Your post was added, #" + data.postId + " but could not be shown.</i>"
+    );
+  }
+}
+
+function errorHandlerForSubmit(err) {
+  const post = $(".post").first();
+  post.before("<p class='warn'>error! " + err + "</p>");
+}
+
```

Comment:
above here, in this file it's just refactoring which I thought would be needed -- though it turned out not to be needed, it results in shorter methods and less nesting